src/webgpu/api/operation/compute/basic.spec.ts

```diff
@@ -160,3 +160,76 @@ g.test('large_dispatch')
 
     dst.destroy();
   });
+
+g.test('buffer_binding_resource')
+  .desc(
+    `Validate the correctness of the buffer binding resource by filling the buffer with
+    testable data, clearing buffer in shader, and verifying the content of the whole buffer:
+  - covers the whole buffer
+  - covers the beginning of the buffer
+  - covers the end of the buffer
+  - covers neither the beginning nor the end of the buffer`
+  )
+  .paramsSubcasesOnly(u =>
+    u //
+      .combine('bindBufferResource', [false, true] as const)
+      .combine('offset', [0, 256, undefined])
+      .combine('size', [4, 8, undefined])
+      .expand('bufferSize', p => [
+        (p.offset ?? 0) + (p.size ?? 16),
+        (p.offset ?? 0) + (p.size ?? 16) + 8,
+      ])
+  )
+  .fn(t => {
+    const { offset, size, bufferSize, bindBufferResource } = t.params;
+
+    const bufferData = new Uint8Array(bufferSize);
```

Comment:
```suggestion
    const bufferData = new Uint32Array(bufferSize / 4);
```

Reply:
optional nit

Reply:
This forces me to add `/ 4` in other places so if that's okay, I'll leave it as is.